contracts/Aavegotchi/facets/WhitelistFacet.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.1;
+
+import {LibMeta} from "../../shared/libraries/LibMeta.sol";
+import {Modifiers, WHITELIST_LIMIT} from "../libraries/LibAppStorage.sol";
+
+contract WhitelistFacet is Modifiers {
+    event WhitelistCreated(uint256 indexed whitelistId, address indexed owner, address[] addresses);
+
+    event WhitelistUpdated(uint256 indexed whitelistId, address indexed owner, address[] addresses);
+
+    function createWhitelist(address[] memory _whitelistAddresses) external {
+        uint256 whitelistLength = _whitelistAddresses.length;
+        require(whitelistLength > 0, "WhitelistFacet: Whitelist length should be larger than zero");
+        require(whitelistLength <= WHITELIST_LIMIT, "WhitelistFacet: Whitelist length exceeds limit");
```

Comment:
Why is a limit to whitelists length needed?

Reply:
Mainly taking into consideration the block gas limit, but there's no strict reason why we'd need an upper limit

Reply:
Understood, that is a good concern to have.   The `updateWhitelist` makes it so that an existing list can be added to, making the possibility of long lists.

Reply:
True, updateWhitelist should also require that the new length is under the limit @orionstardust

Reply:
@orionstardust 
I'm checking new length (sum of old and new array length).

Reply:
Since it's really up to how much gas the user is willing to spend to create their whitelist, I'm just going to remove the whitelist limit.